test/unittest/unittest_nntrainer_models.cpp

```diff
@@ -446,20 +475,36 @@ class nntrainerModelTest
  */
 TEST_P(nntrainerModelTest, model_test) {
   /** Check model with all optimizations off */
-  GraphWatcher g_unopt(getIniName(), false);
-  g_unopt.compareFor(getGoldenName(), getLabelDim(), getIteration());
+  if (getIniName().find("validate") == std::string::npos) {
+    GraphWatcher g_unopt(getIniName(), false);
+    g_unopt.compareFor(getGoldenName(), getLabelDim(), getIteration());
 
-  /// add stub test for tcm
-  EXPECT_EQ(std::get<0>(GetParam()), std::get<0>(GetParam()));
+    /// add stub test for tcm
+    EXPECT_EQ(std::get<0>(GetParam()), std::get<0>(GetParam()));
+  }
 }
 
 /**
  * @brief check given ini is failing/suceeding at load
  */
 TEST_P(nntrainerModelTest, model_test_optimized) {
+  /** Check model with all optimizations on */
+  if (getIniName().find("validate") == std::string::npos) {
```

Comment:
IMO, `*_vallidate` should have different test suite than `model_test` to keep consistency and reduce complexity.

How about we have this validation test added to `unittest_nntrainer_modelfile`? I think this is good spot because other modelfile should have a simplerun test as well.

This isn't directly related to PR. I am willing to work on it after this is merged.

Reply:
A different test suite can be created. However, then each test suite will have to create its own models. If you planning to update test-suite, it would be good if unittests are able to merge and use each other defined models.

The reason for adding the tests here was to validate for now and then later run the same model.
Given this is a valid model, this test wants to check the training/inferencing this model as a user would do it (later add checking exact values).

> How about we have this validation test added to unittest_nntrainer_modelfile? I think this is good spot because other modelfile should have a simplerun test as well.

`unittest_nntrainer_modelfile` provides initialization check only.

IMO these tests can be done later.

Reply:
Understood, if that's the intention I think it explains good reason to leave `*_validate` here